marimo/_runtime/runtime.py

```diff
@@ -1079,6 +1079,36 @@ def instantiate(self, request: CreationRequest) -> None:
             self.run(request.execution_requests)
             self.reset_ui_initializers()
 
+    def handle_message(self, request: ControlRequest) -> None:
+        """Handle a message from the client.
+        The message is dispatched to the appropriate method based on its type.
+        """
+        if isinstance(request, CreationRequest):
+            self.instantiate(request)
+            CompletedRun().broadcast()
+        elif isinstance(request, ExecuteMultipleRequest):
+            self.run(request.execution_requests)
+            CompletedRun().broadcast()
+        elif isinstance(request, SetCellConfigRequest):
+            self.set_cell_config(request)
+        elif isinstance(request, SetUIElementValueRequest):
+            self.set_ui_element_value(request)
+            CompletedRun().broadcast()
+        elif isinstance(request, FunctionCallRequest):
+            status, ret = self.function_call_request(request)
+            FunctionCallResult(
+                function_call_id=request.function_call_id,
+                return_value=ret,
+                status=status,
+            ).broadcast()
+            CompletedRun().broadcast()
+        elif isinstance(request, DeleteRequest):
+            self.delete(request)
+        elif isinstance(request, StopRequest):
+            return None
```

Comment:
`StopRequest` should make the Kernel's control loop quit. This is used for stopping kernels in run mode.